Data/models/DataController.swift

```diff
@@ -3,157 +3,68 @@
+        guard let modelURL = Bundle(for: DataController.self).url(forResource: modelName, withExtension:"momd") else {
             fatalError("Error loading model from bundle")
         }
         guard let mom = NSManagedObjectModel(contentsOf: modelURL) else {
-            fatalError("Error initializing mom from: \(modelURL)")
+            fatalError("Error initializing managed object model from: \(modelURL)")
         }
         
-        self.managedObjectModel = mom
-        self.persistentStoreCoordinator = NSPersistentStoreCoordinator(managedObjectModel: managedObjectModel)
+        let container = NSPersistentContainer(name: modelName, managedObjectModel: mom)
         
-        let urls = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
-        if let docURL = urls.last {
-            do {
-                
-                let options: [String: AnyObject] = [
-                    NSMigratePersistentStoresAutomaticallyOption: true as AnyObject,
-                    NSInferMappingModelAutomaticallyOption: true as AnyObject,
-                    NSPersistentStoreFileProtectionKey : FileProtectionType.complete as AnyObject
-                ]
-                
-                // Old store URL from old beta, can be removed at some point (thorough migration testing though)
-                var storeURL = docURL.appendingPathComponent("Brave.sqlite")
-                try self.persistentStoreCoordinator.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: storeURL, options: options)
-                
-                storeURL = docURL.appendingPathComponent("Model.sqlite")
-                try self.persistentStoreCoordinator.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: storeURL, options: options)
+        // Dev note: This completion handler might be misleading: the persistent store is loaded synchronously by default.
+        container.loadPersistentStores(completionHandler: { _, error in
+            if let error = error {
+                fatalError("Load persistent store error: \(error)")
             }
-            catch {
-                fatalError("Error migrating store: \(error)")
-            }
-        }
-
-        // Setup contexts
-        _ = mainThreadContext
+        })
+        // We need this so the `viewContext` gets updated on changes from background tasks.
+        container.viewContext.automaticallyMergesChangesFromParent = true
+        return container
+    }()
+    
+    private var mainThreadContext: NSManagedObjectContext {
+        return container.viewContext
     }
     
-    public static func remove(object: NSManagedObject, context: NSManagedObjectContext = DataController.shared.mainThreadContext) {
-        context.delete(object)
-        DataController.saveContext(context: context)
+    /// Creates a new background context each time this getter is called.
+    private var backgroundThreadContext: NSManagedObjectContext {
```

Comment:
This is a computed property, not a stored one. So we get new background context each time we call it. Maybe I could rename it to something better to indicate that new context is being created.

Reply:
Ooohh, I see. Yes, this should be renamed, and I think since it always returns a new object, making it a function is probably much more clear. I think we should just name it the same: `newBackgroundContext()`